### ISSUE TITLE
Question about generator model for RTD

### ISSUE THREAD

**Issue by natedingyifeng** [microsoft/CodeBERT#152]
Could you please share more information about the n-gram language model that you have used as the generator model in RTD objective? For example, what is the exact value of n (trigram or 4-gram)? what is the model structure? what are the training hyperparameters?

**Comment by fengzhangyin** [comment]
Hi @natedingyifeng , we use tri-gram (n=3) for the generator model. Instead of using a neural network, we simply use a statistics-based language model. We first calculate the tri-gram distribution on the training data and then predict mask words.

**Comment by natedingyifeng** [comment]
Got it. Thanks!

**Comment by xieexiaotuzi** [comment]
> Hi @natedingyifeng , we use tri-gram (n=3) for the generator model. Instead of using a neural network, we simply use a statistics-based language model. We first calculate the tri-gram distribution on the training data and then predict mask words.

Hi, I am a beginner for "statistics-based language model". I would like to ask why it need to train as mentioned in Section B.1.?
Thanks a lot.

**Comment by fengzhangyin** [comment]
@xieexiaotuzi 
We only train the discriminator of RTD, and the generator is a statistics-based language model without training.

**Comment by xieexiaotuzi** [comment]
> @xieexiaotuzi We only train the discriminator of RTD, and the generator is a statistics-based language model without training.

Thank for your reply. I still have a liitle concern about B.1.1 of CodeBert. As two statistics-based language models are used as generators, why still need time to train?

![image](https://user-images.githubusercontent.com/16114484/221500061-3b85b6dc-b18f-42a2-bb4b-c4fd62e6ff61.png)

Thanks a lot.
